### ISSUE TITLE
Make encodeUser/ decodeUser a Maybe

### ISSUE THREAD

**Issue by amitaibu** [Gizra/elm-restful#10]
https://github.com/Gizra/elm-restful/blob/d27432054b325fd9792db668b4cf74f1059c7988/src/Restful/Login.elm#L510-L511

In most cases I think the user's info will not need to be part of cached credentials, for example - we won't want to save the user's email in the localStorage. For that I suggest to have change these to `Maybe`

```elm
    , decodeUser : Maybe (Decoder user)
    , encodeUser : Maybe (user -> Value)
```

**Comment by amitaibu** [comment]
Actually, you need the `decoder` for when you get the user from `/api/me` - so maybe it should be a setting? e.g. `cacheUser : Bool`

**Comment by rgrempel** [comment]
I guess we could just make the `encodeUser` a `Maybe`. If we don't have it, then we'd just write the `accessToken` to local storage, rather than an encoding of the whole user record. (We'd then have to get the `user` from the backend, which affect the startup process, but I think it's probably doable).

In any event, the basic idea would be to make the local cache of the whole user record optional ... we'll do it if you supply an `encodeUser`, and we'll just cache the `accessToken` if you don't supply an `encodeUser`.

**Comment by amitaibu** [comment]
Sounds good.

**Comment by rgrempel** [comment]
I'm working on this one now.

**Comment by rgrempel** [comment]
Merged in #13.
